### PR TITLE
Campaigns Wizard: referrer segmentation

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [X] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

UI for adding a referrer-based segmentation. 

Do not merge before https://github.com/Automattic/newspack-popups/pull/323)%7C is merged.

### How to test the changes in this Pull Request:

1. Switch Campaign plugin to `add/handle-referrer-segment` branch (https://github.com/Automattic/newspack-popups/pull/323)
1. Create a new segment, observe a new section – "Referrer"
2. Set the referrer to `google.com`
3. Edit or create a campaign, assign the new segment to it
4. Verify the campaign is not displayed when site is visited in a fresh session
5. Spoof the referrer by editing [this line](https://github.com/Automattic/newspack-popups/blob/0493d8db6e3609d95a0409359e1515b5f46259ad/includes/class-newspack-popups-inserter.php#L425) in `class-newspack-popups-inserter.php`
6. Observe that the campaign is displayed now
7. Create a segment with multiple referrers, repeat the testing steps to verify the condition works for all of them
8. Test with a referrer-less segment (just to see if a lacking referrer does not break things)

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [X] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/wizards/popups/views/segmentation/SingleSegment.js [hunk 1]
```diff
@@ -130,16 +126,28 @@ const SegmentsList = ( { segmentId, wizardApiFetch } ) => {
...
-						checked={ is_not_donor }
-						onChange={ setIsNotDonor }
+						checked={ segmentConfig.is_not_donor }
+						onChange={ updateSegmentConfig( 'is_not_donor' ) }
 						label={ __( "Hasn't donated", 'newspack' ) }
 					/>
 				</SegmentSettingSection>
+				<SegmentSettingSection
+					title={ __( 'Referrer', 'newspack' ) }
+					description={ __( 'Segment based on where the traffic is coming from.', 'newspack' ) }
```
Comment (jeffersonrabb) [inline assets/wizards/popups/views/segmentation/SingleSegment.js:141]:
```suggestion
					description={ __( 'Segment based on traffic source.', 'newspack' ) }
```

**Review by jeffersonrabb — APPROVED** [review]
Looks good. Only question I have is about the `www` prefix. I assume most referrers could be either `google.com` or `www.google.com`. If this is true, would we suggest users use referrer lists like `google.com,www.google.com,facebook.com,www.facebook.com`? If so, maybe this should be alluded to, if only in the placeholder for the text input?

**Comment by adekbadek** [comment]
> I assume most referrers could be either `google.com` or `www.google.com`

That's right, how about treating these as equal, since the `www` prefix is so widespread  – a9ad6e58222554f4c7f9e2fa0ce112cb358fe863 – ?

**Comment by adekbadek** [comment]
> I assume most referrers could be either `google.com` or `www.google.com`

That's right, how about treating these as equal, since the `www` prefix is so widespread  – https://github.com/Automattic/newspack-popups/pull/323/commits/94939114b07408e4e9d16b98e985811174e8db94 – ?

**Pull request merged**